flake.nix

```diff
@@ -53,38 +58,76 @@
     };
 
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    pre-commit-hooks = {
+      inputs = {
+        flake-compat.follows = "flake-compat";
+        nixpkgs-stable.follows = "pre-commit-hooks/nixpkgs";
+        nixpkgs.follows = "nixpkgs";
+      };
+
+      url = "github:cachix/pre-commit-hooks.nix";
+    };
+
+    # Interface flake systems.
+    systems.url = "github:nix-systems/default";
   };
 
-  outputs =
-    { nixpkgs, base16, self, ... }@inputs:
-    {
-      packages = nixpkgs.lib.genAttrs [
-        "aarch64-darwin"
-        "aarch64-linux"
-        "i686-linux"
-        "x86_64-darwin"
-        "x86_64-linux"
-      ] (
-        system:
-        let
-          inherit (nixpkgs) lib;
-          pkgs = nixpkgs.legacyPackages.${system};
+  outputs = {
+    nixpkgs,
+    base16,
+    self,
+    ...
+  } @ inputs:
+    inputs.flake-utils.lib.eachDefaultSystem (
+      system: let
+        inherit (nixpkgs) lib;
+        pkgs = nixpkgs.legacyPackages.${system};
+      in {
+        checks.pre-commit-hooks = inputs.pre-commit-hooks.lib.${system}.run {
+          hooks = {
+            alejandra.enable = true;
+            deadnix.enable = true;
+            hlint.enable = true;
+            statix.enable = true;
+            stylish-haskell.enable = true;
+            typos.enable = true;
+            yamllint.enable = true;
+          };
+
+          src = ./.;
+        };
+
+        devShells.default = pkgs.mkShell {
+          inherit (inputs.self.checks.${system}.pre-commit-hooks) shellHook;
 
+          packages = [
+            inputs.home-manager.packages.${system}.default
+            inputs.self.checks.${system}.pre-commit-hooks.enabledPackages
+            pkgs.ghc
```

Comment:
I don't think GHC is necessary since the palette generator should be built using `nix build .#palette-generator`